### PR TITLE
Added SDL2 support

### PR DESCRIPTION
**Pull request by chrisnew**
I opened VS Code today and I found this one. I can’t remember when and why I did it, but I did it: SDL2 support.

### CODE REVIEW

File: src/linux/rw_sdl.c [hunk 1]
```diff
@@ -468,24 +461,33 @@ static qboolean SDLimp_InitGraphics( qboolean fullscreen )
...
+	glcontext = SDL_GL_CreateContext(window);
+#endif
+
+	SDL_GetWindowSize(window, &windowWidth, &windowHeight);
+
+	vid.width = windowWidth;
+	vid.height = windowHeight;
 
-	SDL_WM_SetCaption(APPLICATION, APPLICATION);
+	// SDL_EnableKeyRepeat(SDL_DEFAULT_REPEAT_DELAY, SDL_DEFAULT_REPEAT_INTERVAL); // FIXME: no longer available
```
Comment (jdolan) [inline src/linux/rw_sdl.c:486]: Does this impact anything but typing chat messages?

File: src/linux/rw_sdl.c [hunk 2]
```diff
@@ -468,24 +461,33 @@ static qboolean SDLimp_InitGraphics( qboolean fullscreen )
...
 
-	SDL_EnableKeyRepeat(SDL_DEFAULT_REPEAT_DELAY,
-			SDL_DEFAULT_REPEAT_INTERVAL);
+	SetSDLIcon(); /* currently uses q2icon.xbm data */
 
-	SDL_ShowCursor(0);
+	SDL_SetRelativeMouseMode(SDL_TRUE);
```
Comment (jdolan) [inline src/linux/rw_sdl.c:490]: 👍

File: src/linux/rw_sdl.c [hunk 3]
```diff
@@ -587,9 +589,13 @@ void SWimp_SetPalette( const unsigned char *palette )
 
 void SWimp_Shutdown( void )
 {
-	if (surface)
-		SDL_FreeSurface(surface);
-	surface = NULL;
+	if (window) {
+		SDL_DestroyWindow(window);
+#ifdef GL_QUAKE
```
Comment (jdolan) [inline src/linux/rw_sdl.c:594]: Does this actually need to be conditionalized? Is there a non-GL_QUAKE path?

**Review by jdolan — APPROVED** [review]
Hah! Awesome. Thank you!

**Comment by jdolan** [comment]
@chrisnew Are you in our Discord, btw? There's a ton of activity happening on [https://github.com/jdolan/quetoo](Quetoo).

File: src/linux/rw_sdl.c (hunk 3 above)
Comment (chrisnew) [reply src/linux/rw_sdl.c:594]: good catch, it’s obsolete code path and I just ran with it :sweat_smile:

**Pull request merged**